pyrsistent/_pmap.py

```diff
@@ -125,7 +125,20 @@ def __len__(self):
     def __repr__(self):
         return 'pmap({0})'.format(str(dict(self)))
 
-    __eq__ = Mapping.__eq__
+    def __eq__(self, other):
+        if self is other:
+            return True
+        try:
+            self_hash = hash(self)
+            other_hash = hash(other)
+        except:
```

Comment:
Probably. Fixed.

Although... really hashing is not required for equating things, so regardless what error we get we should not raise it up.... Nevertheless by spec, well-behaving hash implementations should only ever raise TypeErrors or return an int, let's not hide people's mistakes.